### PR TITLE
build kube-cross, go-runner, releng-ci with golang 1.22.8, 1.23.2

### PR DESCRIPTION
**Pull request by haitch**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

- Build Go 1.22.8 images for currently supported Kubernetes release branches
- Build Go 1.23.2 images for currently supported Kubernetes release branches

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/release/issues/3778

#### Does this PR introduce a user-facing change?

```release-note
Build Go 1.23.2 and 1.22.8 images
```

### CODE REVIEW

**Comment by jeremyrickard** [comment]
Let me look at the failure here more and see if this is legit

**Comment by haitch** [comment]
we are blocked on build cross image, it seems [docker.io/library/golang](http://docker.io/library/golang) no longer offer linux/ppc64le linux/s390x arch on bullseye
image

**Comment by dims** [comment]
cc @mkumatag @sudip-ibm @madeelibm for linux/ppc64le linux/s390x arch(es)

**Comment by mkumatag** [comment]
> cc @mkumatag @sudip-ibm @madeelibm for linux/ppc64le linux/s390x arch(es)

Debian bullseye release recently got [EOLed](https://wiki.debian.org/DebianReleases) as per the Debian document and into LTS mode.

image

What really this means that

[Debian Long Term Support (LTS)](https://wiki.debian.org/LTS) is a project to extend the lifetime of all Debian stable releases to (at least) 5 years. Debian LTS is not handled by the Debian Security and Release teams, but by a separate group of volunteers and companies interested in making it a success.

Thus the Debian LTS team takes over security maintenance of the various releases once the Debian Security team stops its work.

[Important](https://wiki.debian.org/LTS/Using): The current LTS version is Debian 11 ("bullseye") and will be supported until August 31st, 2026. Supported architectures in Debian 11 LTS are limited to **_amd64, i386, arm64 and armhf_**. Users of other architectures are especially encouraged to [upgrade to Debian 12 (''bookworm'')](https://wiki.debian.org/DebianUpgrade).

Considering above information and recommendation, lets stop building the bullseye based image for these unsupported platforms and start consuming the later versions. Thanks.

File: images/build/cross/Makefile
Comment (mkumatag) [inline images/build/cross/Makefile:28]: can we remove these arches from `linux/ppc64le linux/s390x` lines from this arch or we can also feed the platforms variable to the variants.yaml to build only on the supported platform.

for testing I have removed these problematic platforms and build went through here - https://github.com/kubernetes/release/pull/3782

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: images/build/cross/Makefile
Comment (haitch) [reply images/build/cross/Makefile:28]: thanks, I can do that. 

shall we wait for [this proposal](https://groups.google.com/a/kubernetes.io/g/dev/c/LBlLCWdFN-o/m/YtGXW-qZBAAJ) to be commonly agreed on?

Comment (haitch) [reply images/build/cross/Makefile:28]: anyway, carried your commit, I guess that PR approve will have to take consideration of that proposal approve.

**Review by cpanato — COMMENTED** [review]
it LGTM

lets wait a bit for some consensus

/hold

**Comment by xmudrii** [comment]
/hold
For the lazy consensus at least until Monday, October 14, 2024, EOD
cc @kubernetes/release-managers @kubernetes/sig-release-leads

**Comment by dims** [comment]
LGTM

**Review by saschagrunert — APPROVED** [review]

**Review by cpanato — APPROVED** [review]
/approve
/lgtm

File: images/build/cross/Makefile
Comment (xmudrii) [inline images/build/cross/Makefile:56]: Nit: can we comment those platforms instead, like we did for `linux/arm`?

**Review by xmudrii — APPROVED** [review]
Tiny non-blocking nit, we can merge without it.
/lgtm
/approve

**Comment by haitch** [comment]
/unhold 

we have the new announcement here: https://groups.google.com/g/kubernetes-sig-release/c/LgxPv6PM4QU/m/SGEv6U_wCQAJ?pli=1

**Comment by xmudrii** [comment]
Timeout error
/retest

**Pull request merged**
